src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,196 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private MyNode<T> first;
+    private MyNode<T> last;
+
     @Override
     public void add(T value) {
+        MyNode<T> newNode = new MyNode<>(null, value, null);
+        if (isEmpty()) {
+            first = newNode;
+        } else {
+            newNode.prev = last;
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkElementIndex(index);
+        MyNode<T> current = findNodeByIndex(index);
+        MyNode<T> newNode = new MyNode<>(null, value, null);
+        if (current.prev == null) {
+            first = newNode;
+        } else {
+            newNode.prev = current.prev;
+            current.prev.next = newNode;
+        }
+        newNode.next = current;
+        current.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        MyNode<T> node = findNodeByIndex(index);
```

Comment:
add checkElementIndex() in findNodeByIndex() and remove from this method

Reply:
```
        if (node == null) {
            return;
        }
        size--;
```

I can't move line "size--;" to the bottom. It's ruin my code.